cloudinit/config/cc_power_state_change.py

```diff
@@ -161,20 +168,39 @@ def load_power_state(cfg):
             (','.join(opt_map.keys()), mode))
 
     delay = pstate.get("delay", "now")
-    # convert integer 30 or string '30' to '+30'
-    try:
-        delay = "+%s" % int(delay)
-    except ValueError:
-        pass
-
-    if delay != "now" and not re.match(r"\+[0-9]+", delay):
-        raise TypeError(
-            "power_state[delay] must be 'now' or '+m' (minutes)."
-            " found '%s'." % delay)
-
-    args = ["shutdown", opt_map[mode], delay]
-    if pstate.get("message"):
-        args.append(pstate.get("message"))
+    if distro_name == 'alpine':
+        # Convert integer 30 or string '30' to '1800' (seconds) as Alpine's
+        # halt/poweroff/reboot commands take seconds as param with no "+"
+        try:
+            delay = "%s" % int(int(delay) * 60)
+        except ValueError:
+            pass
+
+        if delay != "now" and not re.match(r"[0-9]", delay):
+            raise TypeError(
+                "power_state[delay] must be 'now' or '+m' (minutes)."
+                " found '%s'." % delay)
+
+        if delay == "now":
+            # Alpine's halt/poweroff/reboot don't understand "now"
+            delay = "0"
+
+        args = [mode, "-d", delay]
+    else:
+        # convert integer 30 or string '30' to '+30'
+        try:
+            delay = "+%s" % int(delay)
+        except ValueError:
+            pass
+
+        if delay != "now" and not re.match(r"\+[0-9]+", delay):
+            raise TypeError(
+                "power_state[delay] must be 'now' or '+m' (minutes)."
+                " found '%s'." % delay)
+
+        args = ["shutdown", opt_map[mode], delay]
+        if pstate.get("message"):
+            args.append(pstate.get("message"))
```

Comment:
There's a lot of duplicate code here, let's minimize that
```
        def convert_delay(delay, fmt=None, scale=None):
            if not fmt:
                fmt = "+%s"
            if not scale:
                scale = 1

            try:
                delay = fmt % int(int(delay) * int(scale))
            except ValueError
                pass

            if delay != "now" and not re.match(r"[0-9]", delay):
                raise TypeError(
                    "power_state[delay] must be 'now' or '+m' (minutes)."
                    " found '%s'." % delay)

            return delay


        if distro_name == 'alpine':
            # Convert integer 30 or string '30' to '1800' (seconds) as Alpine's
            # halt/poweroff/reboot commands take seconds as param with no "+"
            if delay == 'now':
                delay = "0"
            else:
                delay = convert_delay(delay, fmt="%s", scale=60)
            args = [mode, "-d", delay]
        else
            delay = convert_delay(delay, fmt="+%s", scale=1)
            args = ["shutdown", opt_map[mode], delay]
```

Reply:
Done. I also reworked the re.match regex pattern as it was not properly enforcing the "+minutes" format. I also tweaked the summary section's description of "delay" as it indicated other forms of delay that the shutdown command can handle (such as HH:MM) were valid which conflicted with the rest of the docs which only indicate valid values of 'now' or '+MINUTES'.